### PR TITLE
imgtool: add option for confirming a padded image

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add imgtool command line option for confirming (setting image_ok = 0x01) in a padded image.

Fixes: #664

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

### CODE REVIEW

File: scripts/imgtool/main.py [hunk 1]
```diff
@@ -235,6 +235,8 @@ def convert(self, value, param, ctx):
                    'to 128)')
 @click.option('--pad', default=False, is_flag=True,
               help='Pad image to --slot-size bytes, adding trailer magic')
+@click.option('--confirm', default=False, is_flag=True,
```
Comment (utzig) [inline scripts/imgtool/main.py]: Please just move this new option above `--pad` so it shows afterwards in the help page.

File: scripts/imgtool/image.py [hunk 2]
```diff
@@ -432,8 +433,10 @@ def pad_to(self, size):
                                    self.overwrite_only, self.enckey,
                                    self.save_enctlv, self.enctlv_len)
         padding = size - (len(self.payload) + tsize)
-        pbytes = bytes([self.erased_val] * padding)
-        pbytes += bytes([self.erased_val] * (tsize - len(boot_magic)))
+        pbytes = bytearray([self.erased_val] * padding)
+        pbytes += bytearray([self.erased_val] * (tsize - len(boot_magic)))
+        if self.confirm and not self.overwrite_only:
+            pbytes[-MAX_ALIGN] = 0x01 # image_ok = 0x01
```
Comment (utzig) [inline scripts/imgtool/image.py]: flake8 shows me `E261: at least two spaces before inline comment`

File: scripts/imgtool/main.py (hunk 1 above)
Comment (henrikbrixandersen) [reply scripts/imgtool/main.py]: Fixed.

File: scripts/imgtool/image.py (hunk 2 above)
Comment (henrikbrixandersen) [reply scripts/imgtool/image.py]: Fixed.

**Pull request merged**